Update TESTING.asciidoc for new :qa:os test style

williamrandolph opened the pull request:
The TESTING.asciidoc document had gotten out of date due to some new and
wonderful changes in our vagrant testing code. I've taken a first cut at
some updates here.

Ryan and co. have done a lot to move coordinating code from various little scripts into Gradle tasks and plugins, so we have lost some of the "granularity" we used to have.

I'm sure I'll be cleaning this up a little once I see what it looks like in GitHub.

williamrandolph committed:
Update testing document for new packaging tests

The TESTING.asciidoc document had gotten out of date due to some new and
wonderful changes in our vagrant testing code. I've taken a first cut at
some updates here.

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Build)

williamrandolph committed:
Correct indentation, spelling, and capitalization

williamrandolph commented:
@elasticmachine run elasticsearch-ci/bwc

williamrandolph commented:
@elasticmachine run elasticsearch-ci/bwc

mark-vieira commented on TESTING.asciidoc:
```diff
@@ -313,39 +321,28 @@ a bit more performance out of the process:
+-------------------------------------------------------------------------------------------------
+
+Note that if you interrupt Gradle in the middle of running these tasks, any boxes started
 will remain running and you'll have to stop them manually with `./gradlew stop` or
```
This should be `./gradlew --stop`. @rjernst I assume it's the reaper process that goes back and kills these VMs when the daemon is killed?

rjernst replied:
Yes the reaper will shut the VMs down.

mark-vieira commented on TESTING.asciidoc:
```diff
@@ -438,84 +433,51 @@ that'd consume a ton of ram.
+-----------------------------------------------------------
 
-or on Windows, from a terminal running as Administrator
+If you need to work on the packaging test framework or run Gradle tests from
```
Yeah, this bit is pretty goofy. If this is something we expect folks to do, we should bake this capability into the plugin and provide a task for folks just to spin up a "ready-to-go" VM.

rjernst replied:
This should be possible, but we should give sufficient large warnings/notification that the task is going to leave the VM running, and other guards (eg nothing should depend on such a task, it should only be run directly). A followup issue for discussion/impl details seems fine.

williamrandolph replied:
Cool — I'll remove this section but make a new issue for team discussion. Thanks.

williamrandolph replied:
I've written a follow-up ticket for discussion: https://github.com/elastic/elasticsearch/issues/48087

mark-vieira commented on TESTING.asciidoc:
```diff
@@ -657,7 +619,7 @@ care.
 == Test coverage analysis
```
I'm wonder if we can remove this whole section. Test coverage analysis is not part of our development process in any formal way. This assumes that contributors want some way of analyzing test coverage themselves? I'm not sure just saying "you're on your own" is even worth mentioning.

rjernst replied:
Let's leave that for a separate issue, since this PR is about packaging test docs.

rjernst commented on TESTING.asciidoc:
```diff
@@ -313,39 +321,28 @@ a bit more performance out of the process:
 +
-. Validate your installed dependencies:
+
+. You can run all of the OS packaging tests with `./gradlew distroTest`. This
```
This isn't quite true. `distroTest` is all of the newer java based tests, but does not include the legacy bats tests. To run both, the `packagingTest` task must be used.

williamrandolph replied:
Ah, I didn't know that. Thank you for the correction.

rjernst commented on TESTING.asciidoc:
```diff
@@ -438,84 +433,51 @@ that'd consume a ton of ram.
-sudo bats $BATS_TESTS/20_tar_package.bats $BATS_TESTS/25_tar_plugins.bats
---------------------------------------------
+Because our packaging tests are capable of testing many combinations of OS (Windows
+or Linux), package type (zip file or RPM), Elasticsearch distribution type (default
```
I would use `e.g.` with these parentheticals and maybe `etc` as well, as zip and RPM are not the only package types.

rjernst reviewed (approved):
Looks good, I have a few minor suggestions.

williamrandolph committed:
Merge branch 'master' into packaging-test-doc-update

williamrandolph committed:
Respond to PR feedback

I've edited a few instructions for style, clarity, and correctness.
Furthermore, I've removed the unwieldy instructions for mimicking the
Gradle plugin using shell commands.

williamrandolph committed:
Add some etc's to examples

williamrandolph committed:
Merge branch 'master' into packaging-test-doc-update

williamrandolph commented:
@elasticmachine run elasticsearch-ci/packaging-sample

williamrandolph commented:
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

williamrandolph committed:
Merge branch 'master' into packaging-test-doc-update
